app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+            }
+        }
+
+        if ($this->option('dev')) {
+            $this->warn('The laravel installation will use the latest developmental branch by passing in --dev');
+            $branch = ' --dev';
+        } else {
+            $branch = '';
+        }
+
+        $command = "laravel new {$this->projectname}$branch";
+
+        $this->info("Creating a new project named {$this->projectname}");
+        $this->info("Executing command '$command' in directory {$this->basepath}");
+
+        $process = new Process($command);
+        $process->setWorkingDirectory($this->basepath);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        // @todo Determine why the above outputs before this point; so the following getOutput() call does nothing
+        $this->line($process->getOutput());
+
+        if ($this->replaceEnvVariables()) {
+            $this->info('I replaced .env variables in your new Laravel application');
+        }
+
+        if ($this->option('auth')) {
+            $this->doAuth();
+        }
+
+        if ($this->option('createdb')) {
+            $this->chooseDatabase($this->option('createdb'));
+        }
+
+        if ($this->option('node')) {
+            $this->doNodeOrYarn();
+        }
+
+        if ($this->hasTool('valet')) {
+            if ($this->option('link')) {
+                $this->doValetLink();
+            }
+            $this->openBrowser();
+        }
+
+        if ($this->hasTool('git')) {
+            $this->doGit();
+        }
+
+        $this->openTextEditor();
+
+        $this->info("You're ready to go! Remember to cd into '{$this->projectpath}' before you start editing.");
+    }
+
+
+    protected function setup()
+    {
+        $this->projectname = $this->argument('name');
+        $this->projecturl = 'http://' . $this->projectname . '.dev'; // @todo .dev is no longer good
+
+        $this->basepath = $this->cwd;
+        if ($this->option('path')) {
+            $path = $this->option('path');
+            if (is_dir($path)) {
+                $this->basepath = $path;
+            } else {
+                $this->warn("Your defined '--path $path' is not a directory, so I am skipping it and using '{$this->basepath}' instead.");
+            }
+        }
+
+        $this->projectpath = $this->basepath . DIRECTORY_SEPARATOR . $this->projectname;
+    }
+
+    protected function askToAndRemoveProject()
+    {
+        $this->info("The directory '{$this->projectpath}' already exists.");
+
+        if ($this->confirm("Shall I proceed by removing the following directory? {$this->projectpath}")) {
+            $fs = new Filesystem;
+
+            // @todo Use laravel --force here instead of deleteDirectory()?
+            if ($fs->deleteDirectory($this->projectpath)) {
+                $this->info("I removed the following directory: {$this->projectpath}");
+
+                return true;
```

Comment:
Instead of returning booleans and testing, is there any other Laravel-Zero-friendly way to quit gracefully? Throwing an exception, maybe, or something like Laravel's `abort()`?

Reply:
I don't see such an option (yet?) but see `exit(0)` usage in the Laravel Zero code, so perhaps we can do that here instead of return? Perhaps @nunomaduro has thoughts on the matter?